streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -262,7 +262,7 @@ public void close() {
         if (eosEnabled) {
             streamsProducer.abortTransaction();
         }
-        streamsProducer.close();
+        streamsProducer.flush();
```

Comment:
I'm wondering if we are introducing a latency regression without EOS here: in the old code when closing without EOS we actually do nothing, and now we would block on flushing. 

On the other hand, flushing maybe needed when we close a task to make sure all the tasks' records are acked already.

If the task is in RUNNING before shutting down, we would always commit before closing, so `flush` is already called; if the task is in RESTORING / SUSPENDED there's nothing written from this task, so a `flush` is not needed. So I think it is safe to not call `flush` after all.

Reply:
Sounds reasonable to me

Reply:
Ok, I'll swap this out.